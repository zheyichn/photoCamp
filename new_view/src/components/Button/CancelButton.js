import { Button } from "@mui/material";
import { cancelButtonStyle } from "../../styles/buttonStyles";

export default function CancelButton(props) {
  return (
    <Button
      data-testid="cancel-btn"
      onClick={props.onClick}
      variant="text"
      sx={cancelButtonStyle}
    >
      Cancel
    </Button>
  );
}
