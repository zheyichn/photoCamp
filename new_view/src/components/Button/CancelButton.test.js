import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import CancelButton from "./CancelButton";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";

afterEach(() => {
  cleanup();
});

describe("CancelButton component renders correctly", () => {
  test("cancel button displayed correctly", () => {
    render(<CancelButton />);
    const cancelButton = screen.getByTestId("cancel-btn");
    expect(cancelButton).toBeInTheDocument();
  });

  test("cancel button works appropriately", () => {
    const mockCancelFunction = jest.fn();
    render(<CancelButton onClick={mockCancelFunction} />);
    const cancelButton = screen.getByTestId("cancel-btn");
    userEvent.click(cancelButton);
    expect(mockCancelFunction).toHaveBeenCalled();
  });

  it("cancel button match snapshots", () => {
    const tree = renderer.create(<CancelButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
