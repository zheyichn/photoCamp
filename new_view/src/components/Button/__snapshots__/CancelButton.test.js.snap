// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelButton component renders correctly cancel button match snapshots 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-120lcjt-MuiButtonBase-root-MuiButton-root"
  data-testid="cancel-btn"
  disabled={false}
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  Cancel
</button>
`;
